pkg/eventfilter/attributes/filter.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package attributes
+
+import (
+	"context"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"go.uber.org/zap"
+	"knative.dev/pkg/logging"
+
+	eventingv1beta1 "knative.dev/eventing/pkg/apis/eventing/v1beta1"
+	"knative.dev/eventing/pkg/eventfilter"
+)
+
+type attributesFilter map[string]string
+
+// NewAttributesFilter returns an event filter which performs the exact match on the attributes
+func NewAttributesFilter(attrs map[string]string) eventfilter.Filter {
+	return attributesFilter(attrs)
+}
+
+func (attrs attributesFilter) Filter(ctx context.Context, event cloudevents.Event) eventfilter.FilterResult {
```

Comment:
IMO comment is required for all exported functions. And in this case this function should have comment unless we don't have to export it, which is actually the preferred way.

Reply:
So there are 2 reasons why IMO it doesn't make sense to comment here:

* the receiver is unexported, so user cannot address it directly anyway
* we're implementing a contract and there is nothing, in this particular impl, which deviates particularly from the comment in the contract itself. So what the comment should say? The same comment in eventfilter.Filter? Isn't better to just explicitly show in the code the implementation with the type assertions (like i added) so the developer goes to the original contract and checks the updated docs?